### PR TITLE
fix type not defined error when using external types

### PR DESCRIPTION
**Pull request by GiggleLiu**
Fix the issue https://github.com/TensorBFS/TropicalGEMM.jl/issues/3

### CODE REVIEW

File: src/reconstruct_loopset.jl [hunk 1]
```diff
@@ -176,10 +176,7 @@ function add_mref!(
     sptrs::Expr, ls::LoopSet, ar::ArrayReferenceMeta, @nospecialize(_::Type{Ptr{T}}),
     C::Int, B::Int, sp::NTuple{N,Int}, name::Symbol
 ) where {T,N}
-    Tsym::Symbol = get(VectorizationBase.JULIA_TYPES, T) do
-        Symbol(T)
-    end
-    add_mref_ptr!(sptrs, ls, ar, Tsym, C, B, sp, name)
+    add_mref_ptr!(sptrs, ls, ar, T, C, B, sp, name)
```
Comment (chriselrod) [inline src/reconstruct_loopset.jl:179]: The purpose of this method was to provide a function barrier. You can delete it and rename `add_mref_ptr!`

File: src/reconstruct_loopset.jl [hunk 2]
```diff
@@ -190,16 +187,16 @@ function loop_indexes_bit!(ls::LoopSet, ar::ArrayReferenceMeta)
...
-) where {N}
+) where {N,T}
     @assert B ≤ 0 "Batched arrays not supported yet."
     # maybe no change needed? -- optimize common case
     column_major = ntuple(identity, N)
     li = ar.loopedindex;
     if sp === column_major || isone(length(li))
         # don't set `bit` to true if our vector width is ≥ 8
-        ((Tsym === :Bit) && (ls.vector_width[] < 8)) && loop_indexes_bit!(ls, ar)
+        ((T=== :Bit) && (ls.vector_width[] < 8)) && loop_indexes_bit!(ls, ar)
```
Comment (chriselrod) [inline src/reconstruct_loopset.jl:199]: Should now be `T === VectorizationBase.Bit`, not the symbol `:Bit`

**Comment by chriselrod** [comment]
I already made those changes locally, so I could also just commit and push them.

**Comment by GiggleLiu** [comment]
Sounds good

**Comment by chriselrod** [comment]
(Thanks for the PR!) Fixed by:
https://github.com/JuliaSIMD/LoopVectorization.jl/commit/bbed0ce774e4f73a98393637b5ca66c2d9cc26c7

**Comment by GiggleLiu** [comment]
Cheers!

**Pull request closed without merging**